go-controller/hybrid-overlay/pkg/util/util.go

```diff
@@ -0,0 +1,65 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/types"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/factory"
+
+	kapi "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/klog"
+)
+
+// GetHybridOverlayPortName returns the name of the hybrid overlay switch port
+// for a given node
+func GetHybridOverlayPortName(nodeName string) string {
+	return "int-" + nodeName
+}
+
+// IsWindowsNode returns true if the node has been labeled as a Windows node
+func IsWindowsNode(node *kapi.Node) bool {
+	osPlatform, ok := node.Labels[kapi.LabelOSStable]
+	return ok && osPlatform == "windows"
+}
+
+// GetNodeInternalIP returns the first NodeInternalIP address of the node
+func GetNodeInternalIP(node *kapi.Node) (string, error) {
+	for _, addr := range node.Status.Addresses {
+		if addr.Type == kapi.NodeInternalIP {
+			return addr.Address, nil
+		}
+	}
+	return "", fmt.Errorf("failed to read node %q InternalIP", node.Name)
+}
+
+// StartNodeWatch starts a node event handler
+func StartNodeWatch(h types.NodeHandler, wf *factory.WatchFactory) error {
+	_, err := wf.AddNodeHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			node := obj.(*kapi.Node)
+			klog.V(5).Infof("node ADD event for %q", node.Name)
+			h.Add(node)
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			oldNode := oldObj.(*kapi.Node)
+			newNode := newObj.(*kapi.Node)
+			klog.V(5).Infof("node UPDATE event for %q", newNode.Name)
```

Comment:
(I don't know if we log node update events elsewhere, but nodes get updated A LOT, and we always avoided logging node updates in openshift-sdn, unless something relevant had changed)